vacationtime-solution/src/Code.js

```diff
@@ -0,0 +1,195 @@
+}
+
+/**
+ * Adds manager approval column with drop-down options.
+ */
+function managerApproval() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  var frozenRows = sheet.getFrozenRows();
+  var startRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+
+  // Creates approval column.
+  sheet.insertColumnAfter(lastCol);
+  sheet.getRange(frozenRows, COLUMN_NUMBER.APPROVAL)
+      .setValue('APPROVAL');
+
+  // Sets drop-down menu cells in approval column.
+  var approvalColumnRange = sheet.getRange(startRow, COLUMN_NUMBER.APPROVAL,
+      numRows, 1);
+  var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues)
+      .build();
+  approvalColumnRange.setDataValidation(rule);
+  approvalColumnRange.setValue('IN PROGRESS');
+}
+
+/** 
+ * Creates Notified Column.
+ */
+function createNotifiedColumn() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+  var startRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+  
+  // Sets up column properties.
+  sheet.insertColumnAfter(lastCol);
+  sheet.getRange(1, COLUMN_NUMBER.NOTIFIED)
+      .setValue('NOTIFIED STATUS');
+  var notifiedValues = ['NOTIFIED', 'NOT NOTIFIED'];
+  var rule = SpreadsheetApp.newDataValidation().requireValueInList(notifiedValues)
+      .build();
+
+  // Sets column's cells to be drop-down menus.
+  var notifiedColRange = sheet.getRange(startRow, COLUMN_NUMBER.NOTIFIED,
+      lastRow - frozenRows, 1);
+  notifiedColRange.setDataValidation(rule);
+  notifiedColRange.setValue('NOT NOTIFIED'); 
+  return notifiedColRange;
+}
+
+/**
+ * Creates a calendar event for an employee whose vacation
+ * request has been approved.
+ * @param {String} managerEmail Manager's email.
+ * @param {String} employeeName Name of employee.
+ * @param {String} employeeEmail Email of employee.
+ * @param {date} startDate Vacation request start date.
+ * @param {date} endDate Vacation request end date.
+**/
+function createCalEvent(managerEmail, employeeName, employeeEmail,
+                       startDate, endDate) { 
+    var managerCal = CalendarApp.getCalendarById(managerEmail);
+    // Creates a calendar event.
+    var event = managerCal.createEvent('APPROVED VACATION TIME FOR ' +
+        employeeName, startDate, endDate, {
+        description: 'Your vacation time from ' +
+                                            startDate + ' to ' + endDate +
+                                            ' has been approved! Enjoy!',
+        guests: employeeEmail,
+        sendInvites: true,
+    });
+}
+
+/**
+ * Sends e mails to employees whose vacation time request
+ * was NOT approved.
+ * @param {String} employeeEmail Email of employee.
+**/
+function sendEmails(employeeEmail) {
```

Comment:
Make this more specific, sendRejectionEmail and maybe it should take the request days and craft a specific email body.
"Your request for vacation from X to Y was not approved"